### PR TITLE
libcalib: implementation of common calibration library

### PR DESCRIPTION
**Pull request by niewim19**
## Description

This library will have common parts between calibration and correction utilities and will be used primarily by `calibtool` and `libsensc`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: libs/calib/calib.h [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+#define MAGIRON_PARAMS 12
+#define CHAR_SOFTIRON 's'
+#define CHAR_HARDIRON 'h'
+#define SOFTCAL_ROWSPAN 3
+#define SOFTCAL_COLSPAN 3
+#define HARDCAL_ROWSPAN 3
+#define HARDCAL_COLSPAN 1
+
+typedef enum { typeMagmot = 0, typeMagiron } calibType_t;
```
[the remark this answers is not in the document]
Comment (niewim19) [reply libs/calib/calib.h:31]: unresolvable

File: libs/calib/calib.c [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+	FILE *file;
+	int ret;
+
+	file = fopen(path, "r");
+	if (file == NULL) {
+		fprintf(stderr, "calib: cannot open calibration file '%s'\n", path);
+		return -1;
+	}
+
+	ret = -1;
```
Comment (HBuczynski) [inline libs/calib/calib.c:258]: Initial value can be written during declaration.

File: libs/calib/calib.c [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+			break;
+
+		case typeMagmot:
+		default:
+			return;
+	}
+}
+
+
+/* read calibration file pointed by 'path' searching for calibration named `tag` and saving its content to 'cal' */
```
Comment (HBuczynski) [inline libs/calib/calib.c:246]: I think that one comment in header file is enough.

File: libs/calib/calib.c [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+
+static int calib_magmotRead(FILE *file, calib_t *cal)
+{
+	char *line, *name;
+	size_t lineSz;
+	float value = 0;
+	unsigned int params = 0; /* can be easily mislead correct number of wrong parameters, but better than nothing */
+
+	/* scroll to the `magmot` tag */
+	if (calib_file2tag(file, "magmot") != 0) {
```
Comment (HBuczynski) [inline libs/calib/calib.c:102]: The tag value `magmot` can be defined in a header file and other modules can reuse it. What do you think?

File: libs/calib/calib.c [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+
+		case typeMagmot:
+		default:
+			return;
+	}
+}
+
+
+/* read calibration file pointed by 'path' searching for calibration named `tag` and saving its content to 'cal' */
+int calib_readFile(const char *path, calibType_t type, calib_t *cal)
```
Comment (HBuczynski) [inline libs/calib/calib.c:247]: Please check input arguments

File: libs/calib/calib.c [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+
+	if (motor >= NUM_OF_MOTORS || axis >= 3 || param >= 3) {
+		return NULL;
+	}
+
+	return &cal->params.magmot.motorEq[motor][axis][param];
+}
+
+
+static void calib_magmotDefaults(calib_t *cal)
```
Comment (HBuczynski) [inline libs/calib/calib.c:87]:
```suggestion
static inline void calib_magmotDefaults(calib_t *cal)
```

File: libs/calib/calib.c (hunk 4 above)
Comment (niewim19) [reply libs/calib/calib.c]: Good idea! Implemented.

**Review by HBuczynski — APPROVED** [review]

**Pull request merged**